app/utils/dbutil.py

```diff
@@ -3,15 +3,25 @@
 
 class DBUtil:
 
+    instance = None
+    app = None
+
     def __init__(self, app, test):
-        self.app = app
-        self.test = test
-        # if test:
-        #     self.init_test_database()
-        #     self.connect_db = self.connect_test_database
-        # else:
-        self.init_database()
-        self.connect_db = self.connect_database
+        if DBUtil.instance is None:
```

Comment:
As you have decoupled the implementation from the "self.test" variable, the "test" parameter in get_instance() can be removed, as it also should from the constructor. Infact there shouldn't by _any_ parameters in the constructor of DBUtil. access the app.Config globally, and see if you can remove the need to use the app.open_resource by simply opening the file with f = open("db/database.sql", 'r') , then rv.cursor().executescript(f.read())